File: __iOS_BACKUP__/ShapeByte/features/workout/item/data/Item.swift
//
//  Item.swift
//  ShapeByte
//
//  Created by Lang, Stefan [Shape Byte Tech] on 31.07.24.
//

import Foundation

/**
Represents any kind of workout item, such as exercise, breaks, countdown, cooldown, warm up ...
 */

protocol Item: Equatable, Equals {
    var name: String { get }
}

File: shared/src/commonMain/kotlin/de/stefan/lang/shapebyte/features/workout/item/core/data/Item.kt
package de.stefan.lang.shapebyte.features.workout.item.core.data

import de.stefan.lang.core.assets.ImageAsset

/**
Represents any kind of workout item, such as exercise, breaks, countdown, cooldown, warm up ...
 */

interface Item {
    val name: String
}

interface ImageContaining {
    val imageAsset: ImageAsset?
}

object None : Item {
    override val name: String = "None"
}
